runtime/platform.c

```diff
@@ -137,34 +138,28 @@ static uintnat round_up(uintnat size, uintnat align) {
   return (size + align - 1) & ~(align - 1);
 }
 
-long caml_sys_pagesize = 0;
+intnat caml_plat_pagesize = 0;
+intnat caml_plat_mmap_alignment = 0;
 
 uintnat caml_mem_round_up_pages(uintnat size)
 {
-  return round_up(size, caml_sys_pagesize);
+  return round_up(size, caml_plat_pagesize);
 }
 
-#ifdef _WIN32
-#define MAP_FAILED 0
-#endif
+#define Is_page_aligned(size) ((size & (caml_plat_pagesize - 1)) == 0)
 
 #ifdef DEBUG
 static struct lf_skiplist mmap_blocks = {NULL};
 #endif
 
 #ifndef _WIN32
-Caml_inline void safe_munmap(uintnat addr, uintnat size)
-{
-  if (size > 0)
-    munmap((void*)addr, size);
-}
 #endif
 
 void* caml_mem_map(uintnat size, uintnat alignment, int reserve_only)
 {
-  uintnat alloc_sz = caml_mem_round_up_pages(size + alignment);
-  void* mem;
-  uintnat base, aligned_start, aligned_end;
+  CAMLassert(Is_power_of_2(alignment));
+  CAMLassert(Is_page_aligned(size));
```

Comment:
Reading the code again, I don't understand the reason for this assert (new in the PR) that `size` is page-aligned. It does not fail today, but what benefits does this restriction bring?

Reply:
We never want to be mapping partial pages (just for the waste) - I thought that had come up in one or other of the various PRs on this, but I can't find a reference to it. It formalises something which is already the case and I think it simplifies the reasoning slightly for the Windows/Cygwin mapping, in that we know that the size we request is definitely what gets allocated. Sorry, it's a bit hand-wavy, but I'm fairly sure that's in the first commit on its own for a reason! 🙂